### PR TITLE
Make clear: typed arrays can only be copy-constructed from same type

### LINKED ISSUE #7592: Issue with "BigInt64Array() constructor": `new BigInt64Array( new Int32Array(1) )` is not allowed
MDN URL: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt64Array/BigInt64Array

Note: this issue applies to the constructors for all TypedArrays.  I picked this page because the problem is a little easier to describe for `BigInt64`s.

#### What information was incorrect, unhelpful, or incomplete?

All TypedArray constructors claim they can be constructed from "an object of any of the typed array types (such as `Int32Array`)".  If I'm reading the spec correctly, this is the behaviour it expects.  In practice, `BigInt64` and `BigUint64` typed arrays are not compatible with other types (see below for examples).

#### Specific section or headline?

Syntax > Parameters > typedArray > "an object of any of the typed array types (such as `Int32Array`)"

Alternatively, the page is missing a section discussing the deviation between the spec and observed behaviour.

#### What did you expect to see?

The *Syntax > Parameters > typedArray* section of all TypedArray constructors should indicate that `Big*` parameters are not accepted for non-`Big*` constructors, and vice versa.

and/or

A new section should be added discussing this issue.

#### Did you test this? If so, how?

Testing in the Firefox console shows this is always incompatible:

    >> new Int32Array( new BigInt64Array(0) )
    Uncaught TypeError: BigInt64Array elements are incompatible with Int32Array

    >> new Int32Array( new BigInt64Array(1) )
    Uncaught TypeError: BigInt64Array elements are incompatible with Int32Array

    >> new BigInt64Array( new Int32Array(0) )
    Uncaught TypeError: Int32Array elements are incompatible with BigInt64Array

    >> new BigInt64Array( new Int32Array(1) )
    Uncaught TypeError: Int32Array elements are incompatible with BigInt64Array

Testing in node shows that only non-empty TypedArrays cause errors:

    $ node
    Welcome to Node.js v12.21.0.
    Type ".help" for more information.
    > new Int32Array( new BigInt64Array(0) )
    Int32Array(0) []
    > new Int32Array( new BigInt64Array(1) )
    Uncaught TypeError: Cannot convert a BigInt value to a number
        at new Int32Array (<anonymous>)
    > new BigInt64Array( new Int32Array(0) )
    BigInt64Array(0) []
    > new BigInt64Array( new Int32Array(1) )
    Uncaught TypeError: Cannot convert 0 to a BigInt
        at new BigInt64Array (<anonymous>)
    > 

(Chromium's console output is similar to node)

<details>
<summary>MDN Content page report details</summary>

* Folder: `en-us/web/javascript/reference/global_objects/bigint64array/bigint64array`
* MDN URL: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt64Array/BigInt64Array
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/web/javascript/reference/global_objects/bigint64array/bigint64array/index.md
* Last commit: https://github.com/mdn/content/commit/cf607d68522cd35ee7670782d3ee3a361eaef2e4
* Document last modified: 2021-07-20T18:47:57.000Z

</details>

### PR DESCRIPTION
**Pull request by himanshugarg**
#### Summary
The [TypedArray's](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/TypedArray) behave differently when used as constructor args. Added exception.

#### Motivation
Low hanging :mango: esp with partial fix from @sideshowbarker 

#### Supporting details
See Also https://github.com/mdn/content/pull/7597

#### Related issues
Fixes #7592. Fixex #7597.

#### Metadata
- [x] Fixes a typo, bug, or other error

### CODE REVIEW

File: files/en-us/web/javascript/reference/global_objects/bigint64array/bigint64array/index.md [hunk 1]
```diff
@@ -39,7 +39,7 @@ new BigInt64Array(buffer, byteOffset, length);
     `BYTES_PER_ELEMENT`_ bytes, containing zeros.
 - `typedArray`
   - : When called with a `typedArray` argument, which can be an
-    object of any of the typed array types (such as `Int32Array`), the
+    object of either of the {{glossary("bigint")}} typed array types (such as {{JSxRef("BigUInt64Array")}}), the
```
Comment (sideshowbarker) [inline files/en-us/web/javascript/reference/global_objects/bigint64array/bigint64array/index.md:42]:
```suggestion
    object of either of the {{glossary("bigint")}} typed-array types (such as {{JSxRef("BigUInt64Array")}}), the
```

File: files/en-us/web/javascript/reference/global_objects/biguint64array/biguint64array/index.md [hunk 2]
```diff
@@ -39,7 +39,7 @@ new BigUint64Array(buffer, byteOffset, length);
     `BYTES_PER_ELEMENT`_ bytes, containing zeros.
 - `typedArray`
   - : When called with a `typedArray` argument, which can be an
-    object of any of the typed array types (such as `Int32Array`), the
+    object of either of the {{glossary("bigint")}} typed array types (such as {{JSxRef("BigInt64Array")}}), the
```
Comment (sideshowbarker) [inline files/en-us/web/javascript/reference/global_objects/biguint64array/biguint64array/index.md:42]:
```suggestion
    object of either of the {{glossary("bigint")}} typed-array types (such as {{JSxRef("BigInt64Array")}}), the
```

File: files/en-us/web/javascript/reference/global_objects/float32array/float32array/index.md [hunk 3]
```diff
@@ -40,7 +40,7 @@ new Float32Array(buffer, byteOffset, length);
     `BYTES_PER_ELEMENT`_ bytes, containing zeros.
 - `typedArray`
   - : When called with a `typedArray` argument, which can be an
-    object of any of the typed array types (such as `Int32Array`), the
+    object of any of the **non**-{{glossary("bigint")}} typed array types (such as `Int32Array`), the
```
Comment (sideshowbarker) [inline files/en-us/web/javascript/reference/global_objects/float32array/float32array/index.md:43]:
```suggestion
    object of any of the **non**-{{glossary("bigint")}} typed-array types (such as `Int32Array`), the
```

File: files/en-us/web/javascript/reference/global_objects/float64array/float64array/index.md [hunk 4]
```diff
@@ -40,7 +40,7 @@ new Float64Array(buffer, byteOffset, length);
     `BYTES_PER_ELEMENT`_ bytes, containing zeros.
 - `typedArray`
   - : When called with a `typedArray` argument, which can be an
-    object of any of the typed array types (such as `Int32Array`), the
+    object of any of the **non**-{{glossary("bigint")}} typed array types (such as `Int32Array`), the
```
Comment (sideshowbarker) [inline files/en-us/web/javascript/reference/global_objects/float64array/float64array/index.md:43]:
```suggestion
    object of any of the **non**-{{glossary("bigint")}} typed-array types (such as `Int32Array`), the
```

File: files/en-us/web/javascript/reference/global_objects/int16array/int16array/index.md [hunk 5]
```diff
@@ -39,7 +39,7 @@ new Int16Array(buffer, byteOffset, length);
     `BYTES_PER_ELEMENT`_ bytes, containing zeros.
 - `typedArray`
   - : When called with a `typedArray` argument, which can be an object of any
-    of the typed array types (such as `Int32Array`), the
+    of the **non**-{{glossary("bigint")}} typed array types (such as `Int32Array`), the
```
Comment (sideshowbarker) [inline files/en-us/web/javascript/reference/global_objects/int16array/int16array/index.md:42]:
```suggestion
    of the **non**-{{glossary("bigint")}} typed-array types (such as `Int32Array`), the
```

File: files/en-us/web/javascript/reference/global_objects/int32array/int32array/index.md [hunk 6]
```diff
@@ -36,7 +36,7 @@ new Int32Array(buffer [, byteOffset [, length]]);
     `BYTES_PER_ELEMENT`_ bytes, containing zeros.
 - `typedArray`
   - : When called with a `typedArray` argument, which can be an object of any
-    of the typed array types (such as `Int32Array`), the
+    of the **non**-{{glossary("bigint")}} typed array types (such as `Int32Array`), the
```
Comment (sideshowbarker) [inline files/en-us/web/javascript/reference/global_objects/int32array/int32array/index.md:39]:
```suggestion
    of the **non**-{{glossary("bigint")}} typed-array types (such as `Int32Array`), the
```

File: files/en-us/web/javascript/reference/global_objects/int8array/int8array/index.md [hunk 7]
```diff
@@ -37,7 +37,7 @@ new Int8Array(buffer, byteOffset, length);
     `BYTES_PER_ELEMENT`_ bytes, containing zeros.
 - `typedArray`
   - : When called with a `typedArray` argument, which can be an object of any
-    of the typed array types (such as `Int32Array`), the
+    of the **non**-{{glossary("bigint")}} typed array types (such as `Int32Array`), the
```
Comment (sideshowbarker) [inline files/en-us/web/javascript/reference/global_objects/int8array/int8array/index.md:40]:
```suggestion
    of the **non**-{{glossary("bigint")}} typed-array types (such as `Int32Array`), the
```

File: files/en-us/web/javascript/reference/global_objects/typedarray/index.md [hunk 8]
```diff
@@ -83,12 +83,16 @@ Where _TypedArray_ is a constructor for one of the concrete types.
...
+  - : When called with a `typedArray` argument the `typedArray` gets copied
+    into a new typed array. For **non-{{glossary("bigint")}}** `TypedArray`,
+    the `typedArray` parameter can be an object of only the
+    **non**-{{glossary("bigint")}} typed array types (such as
+    {{JSxRef("Int32Array")}}).  Similarly, for **{{glossary("bigint")}}**
+    `TypedArray`, the `typedArray` parameter can be an object of only the
+    {{glossary("bigint")}} typed array types (such as {{JSxRef("BigInt64Array")}}).
+    Each value in `typedArray` is converted to the corresponding type of the
+    constructor before being copied into the new array.  The length of the new
+    typed array will be same as the length of the `typedArray` argument.
```
Comment (sideshowbarker) [inline files/en-us/web/javascript/reference/global_objects/typedarray/index.md:95]:
```suggestion
  - : When called with a `typedArray` argument, the `typedArray` gets copied
    into a new typed array. For a **non-{{glossary("bigint")}}** `TypedArray`,
    the `typedArray` parameter can be an object of only the
    **non**-{{glossary("bigint")}} typed array types (such as
    {{JSxRef("Int32Array")}}).  Similarly, for a **{{glossary("bigint")}}**
    `TypedArray`, the `typedArray` parameter can be an object of only the
    {{glossary("bigint")}} typed array types (such as {{JSxRef("BigInt64Array")}}).
    Each value in `typedArray` is converted to the corresponding type of the
    constructor before being copied into the new array.  The length of the new
    typed array will be same as the length of the `typedArray` argument.
```

File: files/en-us/web/javascript/reference/global_objects/uint16array/uint16array/index.md [hunk 9]
```diff
@@ -35,7 +35,7 @@ new Uint16Array(buffer, byteOffset, length);
     `BYTES_PER_ELEMENT`_ bytes, containing zeros.
 - `typedArray`
   - : When called with a `typedArray` argument, which can be an object of any
-    of the typed array types (such as `Int32Array`), the
+    of the **non**-{{glossary("bigint")}} typed array types (such as `Int32Array`), the
```
Comment (sideshowbarker) [inline files/en-us/web/javascript/reference/global_objects/uint16array/uint16array/index.md:38]:
```suggestion
    of the **non**-{{glossary("bigint")}} typed-array types (such as `Int32Array`), the
```

File: files/en-us/web/javascript/reference/global_objects/uint32array/uint32array/index.md [hunk 10]
```diff
@@ -39,7 +39,7 @@ new Uint32Array(buffer, byteOffset, length);
     `BYTES_PER_ELEMENT`_ bytes, containing zeros.
 - `typedArray`
   - : When called with a `typedArray` argument, which can be an object of any
-    of the typed array types (such as `Int32Array`), the
+    of the **non**-{{glossary("bigint")}} typed array types (such as `Int32Array`), the
```
Comment (sideshowbarker) [inline files/en-us/web/javascript/reference/global_objects/uint32array/uint32array/index.md:42]:
```suggestion
    of the **non**-{{glossary("bigint")}} typed-array types (such as `Int32Array`), the
```

File: files/en-us/web/javascript/reference/global_objects/uint8array/uint8array/index.md [hunk 11]
```diff
@@ -37,7 +37,7 @@ new Uint8Array(buffer, byteOffset, length);
     `BYTES_PER_ELEMENT`_ bytes, containing zeros.
 - `typedArray`
   - : When called with a `typedArray` argument, which can be an object of any
-    of the typed array types (such as `Int32Array`), the
+    of the **non**-{{glossary("bigint")}} typed array types (such as `Int32Array`), the
```
Comment (sideshowbarker) [inline files/en-us/web/javascript/reference/global_objects/uint8array/uint8array/index.md:40]:
```suggestion
    of the **non**-{{glossary("bigint")}} typed-array types (such as `Int32Array`), the
```

File: files/en-us/web/javascript/reference/global_objects/uint8clampedarray/uint8clampedarray/index.md [hunk 12]
```diff
@@ -39,7 +39,7 @@ new Uint8ClampedArray(buffer, byteOffset, length);
     `BYTES_PER_ELEMENT`_ bytes, containing zeros.
 - `typedArray`
   - : When called with a `typedArray` argument, which can be an object of any
-    of the typed array types (such as `Int32Array`), the
+    of the **non**-{{glossary("bigint")}} typed array types (such as {{JSxRef("Int32Array")}}), the
```
Comment (sideshowbarker) [inline files/en-us/web/javascript/reference/global_objects/uint8clampedarray/uint8clampedarray/index.md:42]:
```suggestion
    of the **non**-{{glossary("bigint")}} typed-array types (such as {{JSxRef("Int32Array")}}), the
```

**Comment by himanshugarg** [comment]
Resolved. Thanks

**Review by sideshowbarker — APPROVED** [review]
Beautiful.

**Pull request merged**